libutil/dr_config.c

```diff
@@ -1495,6 +1495,73 @@ dr_registered_process_iterator_stop(dr_registered_process_iterator_t *iter)
 
 #endif /* WINDOWS */
 
+dr_config_status_t
+dr_register_inject_paths(const char *process_name, process_id_t pid, bool global,
+                         dr_platform_t dr_platform, const char *dr_lib_path,
+                         const char *dr_alt_lib_path)
+{
+#ifdef PARAMS_IN_REGISTRY
+    ConfigGroup *policy = get_policy(dr_platform);
+    ConfigGroup *proc_policy = get_proc_policy(policy, process_name);
+#else
+    FILE *f = open_config_file(process_name, pid, global, dr_platform, true /*read*/,
+                               true /*write*/, false /*!overwrite*/);
+#endif
+    dr_config_status_t status = DR_SUCCESS;
+    TCHAR wpath[MAXIMUM_PATH];
+    if (IF_REG_ELSE(proc_policy == NULL, f == NULL)) {
+        status = DR_PROC_REG_INVALID;
+        goto exit;
+    }
+    if (dr_lib_path == NULL) {
+        status = DR_CONFIG_INVALID_PARAMETER;
+        goto exit;
+    }
+#ifndef PARAMS_IN_REGISTRY
+    /* shift rest of file up, overwriting old value, so we can append new value */
+    read_config_ex(f, DYNAMORIO_VAR_AUTOINJECT, NULL, 0, true);
+#endif
+    convert_to_tchar(wpath, dr_lib_path, MAXIMUM_PATH);
+    NULL_TERMINATE_BUFFER(wpath);
+    status = write_config_param(IF_REG_ELSE(proc_policy, f),
+                                PARAM_STR(DYNAMORIO_VAR_AUTOINJECT), wpath);
+    DO_ASSERT(status == DR_SUCCESS);
+    if (status != DR_SUCCESS)
```

Comment:
`if (status != DR_SUCCESS)` We assert with on `DO_ASSERT(status == DR_SUCCESS)` so if that assert passes, there won't be a need to check `if status != DR_SUCCESS`, right?

Reply:
This does seem odd: it's a pattern elsewhere in the file.  Since it handles the failure case by returning status it seems the assert should be removed.  Probably it was meant as a something like core's ASSERT_CURIOSITY.  I'm removing the assert.